### PR TITLE
Feat/tag

### PR DESCRIPTION
**Pull request by ahmadanwar93**
- Tag component 
- Tag stories
- Extend Tailwind merge as per this issue (https://github.com/dcastil/tailwind-merge/issues/297)

### CODE REVIEW

File: packages/react/src/components/tag.tsx
Comment (irfancoder) [inline packages/react/src/components/tag.tsx:1]: 1. Missing border
2. Check if the dot grows according to size

File: packages/react/src/components/tag.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      },
+      size: {
+        small: "py-0.5 px-2 text-body-xs h-[22px]",
+        medium: "py-1 px-2 text-body-sm h-[28px]",
+        large: "py-1 px-2.5 text-body-md h-8",
+      },
+      tagStyle: {
+        style1: "rounded-full",
+        style2: "rounded-md",
+      },
```
Comment (irfancoder) [inline packages/react/src/components/tag.tsx:24]: Rename:
1. ~~tagStyle~~ mode
2. ~~style1~~ pill
3. ~~style2~~ default

**Review by irfancoder — CHANGES_REQUESTED** [review]

File: packages/react/src/components/tag.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+const tag_cva = cva(
+  ["inline-flex gap-[5px] font-medium font-body justify-center items-center"],
+  {
+    variants: {
+      variant: {
+        gray: "text-txt-black-500 bg-bg-washed",
+        brand: "bg-bg-primary-50 text-txt-primary",
+        success: "bg-bg-success-50 text-txt-success",
+        danger: "bg-bg-danger-50 text-txt-danger",
+        warning: "bg-bg-warning-50 text-txt-warning",
```
[the remark this answers is not in the document]
Comment (ahmadanwar93) [reply packages/react/src/components/tag.tsx:14]: Done

File: packages/react/src/components/tag.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * Props for Tag component.
+ * @typedef TagProps
+ * @property {React.ReactNode} children - The content of the tag (text or HTML elements)
+ * @property {boolean} [showStatusDot=false] - Whether to show a status dot
+ * @property {"gray" | "brand" | "success" | "danger" | "warning"} variant - The visual style variant of the tag
+ * @property {"small" | "medium" | "large"} size - The size of the tag
+ * @property {"style1" | "style2"} tagStyle - The style of the tag (rounded or not)
+ */
+interface TagProps extends ComponentProps<"div">, VariantProps<typeof tag_cva> {
+  showStatusDot?: boolean;
```
[the remark this answers is not in the document]
Comment (ahmadanwar93) [reply packages/react/src/components/tag.tsx:44]: Done as well

File: packages/react/src/components/tag.tsx (hunk 1 above)
Comment (ahmadanwar93) [reply packages/react/src/components/tag.tsx:24]: Done as well

File: packages/react/src/components/tag.tsx
Comment (ahmadanwar93) [reply packages/react/src/components/tag.tsx:1]: Done as well

**Review by irfancoder — APPROVED** [review]
Looking good

**Pull request merged**